### ISSUE TITLE
`prevent-abbreviations` should ignore config files

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#2015]
[prevent-abbreviations](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prevent-abbreviations.md) flagged the [config file](https://github.com/bmish/eslint-doc-generator#configuration-file) `.eslint-doc-generatorrc.js` from [eslint-doc-generator](https://github.com/bmish/eslint-doc-generator). I hit this in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2013.

```
/home/runner/work/eslint-plugin-unicorn/eslint-plugin-unicorn/.eslint-doc-generatorrc.js
Error:   2:1  error  The filename `.eslint-doc-generatorrc.js` should be named `.eslint-document-generatorrc.js`. A more descriptive name will do too  unicorn/prevent-abbreviations
```

But we can't rename config files as those names are not controlled by us. So we should exempt them.

I would recommend exempting filenames of any of these common formats supported by cosmiconfig: https://github.com/davidtheclark/cosmiconfig#cosmiconfigoptions 

Especially:
* Hidden files starting with `.`
* Any filename ending with `rc` or `.config`

**Comment by silverwind** [comment]
I don't think anyone here wants to maintain a list of config file names, so I suggest you put an override instead:

````yaml
overrides:
  - files: ["**/*.config.js"]
    rules:
      rule: [0]
